src/frontend/src/lib/derived/network-tokens.derived.ts

```diff
@@ -36,11 +41,9 @@ export const enabledErc20NetworkTokens: Readable<Erc20Token[]> = derived(
 // TODO: The several dependencies of enabledIcNetworkTokens are not strictly only IC tokens, but other tokens too.
 //  We should find a better way to handle this, improving the store.
 export const enabledIcNetworkTokens: Readable<IcToken[]> = derived(
```

Comment:
Should this derived store maybe be removed, it isn't used anymore?

If yes, not in this PR.

Reply:
After I move all the things we discussed above, `network-tokens.derived` will have 3 stores left: `enabledIcNetworkTokens` (internal, not exported), `combinedDerivedSortedNetworkTokens` (exported) and `combinedDerivedSortedNetworkTokensUi` (exported). Those ones are filtered with `filterTokensForSelectedNetwork` so we can potentially keep them in that file, unless you think that even with that filtering they belong to `tokens.derived`.